server/dhcp.py

```diff
@@ -92,11 +93,12 @@ def get_option(self, option):
     def options(self):
         done = list()
         # fulfill wishes
-        for option in self.parameter_order:
-            if option < len(options) and hasattr(self, options[option][0]) or hasattr(self, 'option_{}'.format(option)):
-                # this may break with the specification because we must try to fulfill the wishes
-                if option not in done:
-                    done.append(option)
+        if self.parameter_order:
```

Comment:
This change looks like it does not make a difference. Can you tell me what it does and how it helps?

Reply:
This helps in the situation where `parameter_order == None`. Iterating over `None` raises an exception.

Reply:
In that case, it helps. Where is parameter_order set to None?

Reply:
I was dealing with a device that I suspect is not compliant. I assumed that it had provided some invalid request.

I didn't however debug this all the way.